service/src/main/java/bio/terra/landingzone/stairway/flight/create/resource/step/CreateLandingZoneFederatedIdentityStep.java

```diff
@@ -0,0 +1,168 @@
+package bio.terra.landingzone.stairway.flight.create.resource.step;
+
+import bio.terra.landingzone.library.landingzones.definition.ArmManagers;
+import bio.terra.landingzone.service.landingzone.azure.model.LandingZoneResource;
+import bio.terra.landingzone.stairway.common.model.TargetManagedResourceGroup;
+import bio.terra.landingzone.stairway.flight.utils.FlightUtils;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import com.azure.resourcemanager.msi.fluent.models.FederatedIdentityCredentialInner;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.apis.CoreV1Api;
+import io.kubernetes.client.openapi.models.V1DeleteOptions;
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import io.kubernetes.client.openapi.models.V1ServiceAccount;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+
+public class CreateLandingZoneFederatedIdentityStep implements Step {
+  private static final Logger logger =
+      LoggerFactory.getLogger(CreateLandingZoneFederatedIdentityStep.class);
+  public static final String k8sNamespace = "default";
+  private final ArmManagers armManagers;
+  private final KubernetesClientProvider kubernetesClientProvider;
+
+  public CreateLandingZoneFederatedIdentityStep(
+      ArmManagers armManagers, KubernetesClientProvider kubernetesClientProvider) {
+    this.armManagers = armManagers;
+    this.kubernetesClientProvider = kubernetesClientProvider;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    var uami =
+        FlightUtils.getRequired(
+            context.getWorkingMap(),
+            CreateLandingZoneIdentityStep.LANDING_ZONE_IDENTITY_RESOURCE_KEY,
```

Comment:
If I'm understanding -- this is a UAMI created in [CreateLandingZoneIdentityStep](https://github.com/DataBiosphere/terra-landing-zone-service/blob/main/service/src/main/java/bio/terra/landingzone/stairway/flight/create/resource/step/CreateLandingZoneIdentityStep.java) solely for the purpose of administering postgres, right?

I was questioning whether this will conflict with Leo stuff but I don't think so -- Leo should still be able to associate "pet" UAMIs via aad-pod-identity in app namespaces.